fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -40,6 +40,9 @@ def load_data(self, X, y, event=None, **kwargs):
         """
         super().load_data(X, y, **kwargs)
         self.tags[_EVENT] = event
+        self.tags = self.tags.dropna()
+        self.dropped_X = self.X.drop(self.tags[_LABEL].index)
+        self.X = self.X.drop(self.dropped_X.index)
         self.prob_event = self.tags.groupby(_EVENT).size() / self.total_samples
```

Comment:
Do we need to do all this dropping above? I believe that `groupby` automatically drops `np.nan`. also, it would be great if we could avoid transforming `X`.

Reply:
Oh yes, I think I understood what you mean by this. I will check this way and let you know here if it works the same way! Thank you.